Entities/SceneLayer.cpp

```diff
@@ -505,41 +505,42 @@ namespace RTE {
 
 	template <>
 	void SceneLayerImpl<true>::RegisterDrawing(int left, int top, int right, int bottom) {
-		m_Drawings.push_back(IntRect(left, top, right, bottom)); 
+		m_Drawings.emplace_back(left, top, right, bottom); 
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
 	template <>
 	void SceneLayerImpl<false>::RegisterDrawing(int left, int top, int right, int bottom) { /* no-op */ }
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	template <bool TRACK_DRAWINGS>
-	void SceneLayerImpl<TRACK_DRAWINGS>::RegisterDrawing(const Vector& center, float radius)
-	{
+	void SceneLayerImpl<TRACK_DRAWINGS>::RegisterDrawing(const Vector &center, float radius) {
 		if (radius != 0.0F) {
-			RegisterDrawing(center.m_X - radius, center.m_Y - radius, center.m_X + radius, center.m_Y + radius);
+			RegisterDrawing(static_cast<int>(center.m_X - radius), static_cast<int>(center.m_Y - radius), static_cast<int>(center.m_X + radius), static_cast<int>(center.m_Y + radius));
 		}
 	}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	template <>
-	void SceneLayerImpl<false>::ClearDrawings(BITMAP* bitmap, const std::vector<IntRect>& /*drawings*/, ColorKeys clearTo) {
```

Comment:
I understand this and am happy with it either way, though preferentially in the future I want to enable the warning when a qualified parameter is unused. It’s actually fixed some bugs in my experience, though it’s a bit annoying.

Reply:
I've got no issue with that. It's a rare thing to be doing and probably indicative of something questionable.
But yeah, I definitely don't care for params without names like that, generally feels like a mistake to me.